Plan/nukkit/src/main/java/com/djrapitops/plan/addons/placeholderapi/PlaceholderRegistrar.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *  This file is part of Player Analytics (Plan).
+ *
+ *  Plan is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License v3 as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Plan is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Plan. If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.djrapitops.plan.addons.placeholderapi;
+
+import cn.nukkit.Player;
+import com.creeperface.nukkit.placeholderapi.api.PlaceholderAPI;
+import com.djrapitops.plan.PlanSystem;
+import com.djrapitops.plan.delivery.domain.container.PlayerContainer;
+import com.djrapitops.plan.delivery.domain.keys.PlayerKeys;
+import com.djrapitops.plan.gathering.cache.SessionCache;
+import com.djrapitops.plan.placeholder.PlanPlaceholders;
+import com.djrapitops.plan.storage.database.queries.containers.ContainerFetchQueries;
+import com.djrapitops.plugin.logging.L;
+import com.djrapitops.plugin.logging.error.ErrorHandler;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+public class PlaceholderRegistrar {
+
+    private final ErrorHandler errorHandler;
+    private final PlanSystem system;
+
+    public PlaceholderRegistrar(PlanSystem system, ErrorHandler errorHandler) {
+        this.system = system;
+        this.errorHandler = errorHandler;
+    }
+
+    private void register0() {
+        PlanPlaceholders.init(system);
+
+        PlaceholderAPI api = PlaceholderAPI.getInstance();
+        PlanPlaceholders.getPlaceholders().forEach((name, loader) ->
+                api.visitorSensitivePlaceholder(name, (player, params) -> {
+                            try {
+                                return loader.apply(getPlayer(player), new ArrayList<>(params.getAll().values()));
```

Comment:
It's because we need to get List instance to make working with parameters easier (I think I changed that in the future version of PAPI that I'm going to probably release for nukkit 2.0) so currently I don't know about a better way of converting Collection to ArrayList.